include/mlir/Dialect/SPIRV/SPIRVBase.td

```diff
@@ -227,6 +230,20 @@ class SPV_Optional<Type type> : Variadic<type>;
 // TODO(ravishankarm): From 1.4, this should also include Composite type.
 def SPV_SelectType : AnyTypeOf<[SPV_Scalar, SPV_Vector, SPV_AnyPtr]>;
 
+// The execution scope or memory scope of an operation. When used as a memory
+// scope, it specifies the distance of synchronization from the current
+// invocation. When used as an execution scope, it specifies the set of
+// executing invocations taking part in the operation.
+def SPV_Scope : IntOfWidths<[32]>;
```

Comment:
ot sure about this approach. Should really be using an enum to describe all the valid semantics.. The enum would be an I32Attr (see other enum definitions in SPIRVBase.td). You could then let the operation take a list of these attributes (i.e. and ArrayAttr) and then having the instructions take an array of these